File: Solutions/0367- Valid Perfect Square/Solution.py
class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        left, right = 1, num
        while left < right:
            middle = (left + right) // 2
            if middle * middle >= num:
                right = middle
            else:
                left = middle + 1
        return left * left == num

File: Solutions/0367- Valid Perfect Square/Solution.java
class Solution {
    public boolean isPerfectSquare(int num) {
        long left = 1;
        long right = num;
        while (left < right) {
            long middle = (left + right) >>> 1;
            if (middle * middle >= num) {
                right = middle;
            } else {
                left = middle + 1;
            }
        }
        return left * left == num;
    }
}